sw/device/lib/testing/test_framework/BUILD

```diff
@@ -299,35 +299,39 @@ _FLOW_CONTROL_MESSAGE = (
     "Everywhere that Mary went the lamb was sure to go."
 )
 
-opentitan_functest(
+opentitan_test(
     name = "ottf_flow_control_functest",
     srcs = ["ottf_flow_control_functest.c"],
     cw310 = cw310_params(
-        test_cmds = [
-            "--exec=\"fpga load-bitstream $(location {bitstream})\"",
-            "--exec=\"transport init\"",
-            "--exec=\"bootstrap --clear-uart=true $(location {flash})\"",
-            "--exec=\"console --quiet --exit-success=WAIT --exit-failure=PASS|FAIL --flow-control\"",
-            "console",
-            "--flow-control",
-            "--send=\"{}\n\"".format(_FLOW_CONTROL_MESSAGE),
-            "--exit-success=\"RESULT:{}\"".format(_FLOW_CONTROL_MESSAGE),
-            "--exit-failure=\"PASS|FAIL\"",
-        ],
+        flow_control_message = _FLOW_CONTROL_MESSAGE,
+        newline = "\n",
```

Comment:
We are currently discussing what is the best way to handle newline chars in `test_cmd`. The current implementation replaces `\n` by spaces to generate a command in a single line.

Reply:
bazel_skylib's `shell.quote` doesn't do anything that would help us here.

- We can use the current multi-line string scheme, substituting newlines for spaces to generate a single string and requiring special handling when a literal newline is required.
- We can go back to the old scheme: a list of strings gets joined together with spaces and turned into the command line.
- We can use the shell to handle building the argument list and avoid doing the "\n" replacement.  In rules/opentitan/fpga_cw310.bzl, we alter the build template to say `TEST_CMD=({test_cmd})`, which will make the shell build a list variable.  Newlines between arguments will create separate list elements.  Newlines in quotes will be preserved.  We then expand `TEST_CMD` on the command line as `"${TEST_CMD[@]}"`.

I've tried the third option and it works.  Its probably the best option.  I can push the change to this PR if you like.